feat/feature.go

```diff
@@ -67,6 +67,14 @@ type Range interface {
 	Len() int
 }
 
+// Feature is a Range whose coordinates are defined relative to a feature
+// location. Start and End return the coordinates of the feature relative to
+// its location which can be nil. In the latter case callers should make no
+// assumptions whether coordinates of such features are comparable.
+//
+// Name returns the name of the feature.
+// Description returns the description of the feature.
+// Location returns the reference feature where the feature is located.
 type Feature interface {
```

Comment:
Just curious. Any specific reason for this? e.g. I see you had `Orientation()` in the top-level comment and it looks the io package in the standard library has such doc in the top level.